src/Hedgehog/Plutus/Tx.hs

```diff
@@ -104,7 +112,125 @@ data ScriptPurpose
   In the future, hps should support custom balancers.
 -}
 balanceTx :: Model.Mock -> Tx 'Unbalanced -> Maybe (Tx 'Balanced)
-balanceTx = _
+balanceTx mock tx = do
+  let balanceingKeys = Map.keys $ Model.mockUsers mock
```

Comment:
We don't need to know about keys, just take UTxOs until the balance is covered

Reply:
I guess I still don't really understand why we would ever want to balance a transaction with utxos from all users so I wrote it to be easy to convert into balancing as a user or set of users.
If we're going to have happy path testing we will need to be able to balance as a single user or set of users, if we really need both balancing as every user just being a special case of balancing as a set of users feels like a good way to do that.

Reply:
If the script relies on a specific PubKey, this must be described in the test data. If it does not, then we should test that the PubKey doesn't matter - even in happy path.

In my mind, 'happy path' means the script succeeds when it is supposed to - not just in limited cases when we expect it to.